### PR TITLE
fix: failing tests on sync engine WPB-6196

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6196" title="WPB-6196" target="_blank">TaskWPB-6196</a>  [iOS] Fix failing sync engine tests
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Failing tests on sync engine needs fixing

The main reason for tests failing seems to be due to using `WaitingGroupTask` with a managed object context different from the context used by the tests. We wait for all groups to be empty but since we're using different context, the changes are not in sync. As a solution, we can save the context when we're done. 

There seems to be another issue related to Xcode and defining tests within extensions. Not sure what happens but it looks like some failing tests are not always being detected by Xcode.

### CODE REVIEW

**Review by caldrian — APPROVED** [review]

**Comment by netbe** [comment]
> The main reason for tests failing seems to be due to using WaitingGroupTask with a managed object context different from the context used by the tests. 

wait the groups are not shared by contexts? it's one set of groups per context?

**Review by typfel — APPROVED** [review]

**Comment by typfel** [comment]
> > The main reason for tests failing seems to be due to using WaitingGroupTask with a managed object context different from the context used by the tests.
> 
> wait the groups are not shared by contexts? it's one set of groups per context?

They are using the same managed object contexts.

From looking at the changes I was thinking it can be that the code was previously relying on implicit saves which are made after a http response has been processed. Now if during in the processing you do core data changes in a task it would happen after the implicit save.

File: wire-ios-sync-engine/Tests/Source/E2EE/ConversationTests+OTR.m [hunk 1]
```diff
@@ -284,10 +284,7 @@ - (void)testThatItOTRAssetCanExpire
     WaitForAllGroupsToBeEmpty(0.5);
 }
 
-@end
-
 #pragma mark - Trust
-@implementation ConversationTestsOTR (Trust)
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Tests/Source/E2EE/ConversationTests+OTR.m]: Nice 👍

File: wire-ios-sync-engine/Source/Data Model/Conversation+MessageDestructionTimeout.swift [hunk 2]
```diff
@@ -48,15 +48,21 @@ extension ZMConversation {
...
             return completion(.failure(WirelessLinkError.unknown))
         }
 
-        let request = MessageDestructionTimeoutRequestFactory.set(timeout: Int(timeout.rawValue), for: self, apiVersion: apiVersion)
+        let request = MessageDestructionTimeoutRequestFactory.set(
+            timeout: Int(timeout.rawValue),
+            for: self,
+            apiVersion: apiVersion
+        )
+
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Source/Data Model/Conversation+MessageDestructionTimeout.swift]: Linter is complaining here:
> Lines should not have trailing whitespace (trailing_whitespace)

**Review by markusfassbender — APPROVED** [review]

**Review by netbe — APPROVED** [review]

**Pull request closed without merging**